### PR TITLE
GB-1464 bitpanda Pro exchange precision fix

### LINKED ISSUE #407: Bitpanda PRO Exchange
Hi, we get following error: "error_code": "BAD_AMOUNT_PRECISION",

### PR DESCRIPTION
**Pull request by pvyhnal-generalbytes**
Fixes #407

### CODE REVIEW

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitpandapro/BitpandaProExchange.java [hunk 1]
```diff
@@ -84,6 +86,23 @@ public static IExchangeAdvanced asExchange(String apikey, String preferredFiatCu
...
+    // https://api.exchange.bitpanda.com/public/v1/currencies
+    static {
+        fiatPricePrecisions.put("EUR", 2);
+        fiatPricePrecisions.put("CHF", 2);
+
+        instrumentAmountPrecisions.put("BTC_EUR", 5);
+        instrumentAmountPrecisions.put("ETH_EUR", 4);
+        instrumentAmountPrecisions.put("XRP_EUR", 0);
+        instrumentAmountPrecisions.put("BTC_CHF", 5);
+        instrumentAmountPrecisions.put("ETH_CHF", 2);
```
Comment (pyranja) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitpandapro/BitpandaProExchange.java:102]:
```suggestion
        instrumentAmountPrecisions.put("ETH_CHF", 4);
```

File: server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitpandapro/BitpandaProExchange.java [hunk 2]
```diff
@@ -84,6 +86,23 @@ public static IExchangeAdvanced asExchange(String apikey, String preferredFiatCu
...
     );
 
+    static private final Map<String, Integer> fiatPricePrecisions = new HashMap<>();
+    static private final Map<String, Integer> instrumentAmountPrecisions = new HashMap<>();
+
+    // https://api.exchange.bitpanda.com/public/v1/instruments
+    // https://api.exchange.bitpanda.com/public/v1/currencies
+    static {
+        fiatPricePrecisions.put("EUR", 2);
+        fiatPricePrecisions.put("CHF", 2);
```
Comment (pyranja) [inline server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitpandapro/BitpandaProExchange.java:96]: For prices the instrument's market precision is relevant, e.g. for XRP_EUR market precision is, allowing up to 5 decimal places on prices.

Comment (pvyhnal-generalbytes) [reply server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitpandapro/BitpandaProExchange.java:96]: This is addressed in #411, right?

Comment (pyranja) [reply server_extensions_extra/src/main/java/com/generalbytes/batm/server/extensions/extra/bitcoin/exchanges/bitpandapro/BitpandaProExchange.java:96]: yes the fixed price calculation uses prices from the orderbook as is, which have an appropriate precision.

**Comment by pvyhnal-generalbytes** [comment]
fixed in #411

**Pull request closed without merging**